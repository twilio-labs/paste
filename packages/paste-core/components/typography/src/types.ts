export interface ParagraphProps {
  id?: never;
  className?: never;
  children: NonNullable<React.ReactNode>;
}
