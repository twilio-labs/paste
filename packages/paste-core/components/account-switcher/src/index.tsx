export * from "./AccountSwitcher";
export * from "./AccountSwitcherGroup";
export * from "./AccountSwitcherSeparator";
export * from "./AccountSwitcherItem";
export * from "./AccountSwitcherItemRadio";
export * from "./AccountSwitcherBadge";
export * from "./useAccountSwitcher";
