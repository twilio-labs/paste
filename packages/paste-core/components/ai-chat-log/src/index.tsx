export { AIChatMessage } from "./AIChatMessage";
export type { AIChatMessageProps } from "./AIChatMessage";
export { AIChatMessageActionGroup } from "./AIChatMessageActionGroup";
export type { AIChatMessageActionGroupProps } from "./AIChatMessageActionGroup";
export { AIChatMessageBody } from "./AIChatMessageBody";
export type { AIChatMessageBodyProps } from "./AIChatMessageBody";
export { AIChatMessageAuthor } from "./AIChatMessageAuthor";
export type { AIChatMessageAuthorProps } from "./AIChatMessageAuthor";
export { AIChatMessageActionCard } from "./AIChatMessageActionCard";
export type { AIChatMessageActionCardProps } from "./AIChatMessageActionCard";
export { AIChatMessageLoading } from "./AIChatMessageLoading";
export type { AIChatMessageLoadingProps } from "./AIChatMessageLoading";

export { AIChatLog } from "./AIChatLog";
export type { AIChatLogProps } from "./AIChatLog";
export { useAIChatLogger } from "./useAIChatLogger";
export type { UseAIChatLogger, AIPartialIDChat, AIChat } from "./useAIChatLogger";
export { AIChatLogger } from "./AIChatLogger";
export type { AIChatLoggerProps } from "./AIChatLogger";

export type { AIMessageVariants } from "./AIMessageContext";
