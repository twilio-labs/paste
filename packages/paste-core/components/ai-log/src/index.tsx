export { AIChatMessage } from "./AIChatMessage";
export type { AIChatMessageProps } from "./AIChatMessage";
export { AIChatMessageMeta } from "./AIChatMessageMeta";
export type { AIChatMessageMetaProps } from "./AIChatMessageMeta";
export { AIChatMessageMetaItem } from "./AIChatMessageMetaItem";
export type { AIChatMessageMetaItemProps } from "./AIChatMessageMetaItem";
export { AIChatMessageContent } from "./AIChatMessageContent";
export type { AIChatMessageContentProps } from "./AIChatMessageContent";

export { AIChatLog } from "./AIChatLog";
export type { AIChatLogProps } from "./AIChatLog";
export { useAIChatLogger } from "./useAIChatLogger";
export type { UseAIChatLogger } from "./useAIChatLogger";
export { AIChatLogger } from "./AIChatLogger";
export type { AIChatLoggerProps } from "./AIChatLogger";
