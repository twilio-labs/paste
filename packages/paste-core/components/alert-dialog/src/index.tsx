import * as React from 'react';
import * as PropTypes from 'prop-types';
import {useUID} from '@twilio-paste/uid-library';
import {useTransition} from '@twilio-paste/animation-library';
import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
import type {BoxProps} from '@twilio-paste/box';
import {ModalDialogOverlay} from '@twilio-paste/modal';
import {AlertDialogHeader} from './AlertDialogHeader';
import {AlertDialogBody} from './AlertDialogBody';
import {AlertDialogContent} from './AlertDialogContent';
import {AlertDialogFooter} from './AlertDialogFooter';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getAnimationStates = (): any => ({
  from: {opacity: 0, transform: `scale(0.675)`},
  enter: {opacity: 1, transform: `scale(1)`},
  leave: {opacity: 0, transform: `scale(0.675)`},
  // https://www.react-spring.io/docs/hooks/api
  config: {
    mass: 0.5,
    tension: 370,
    friction: 26,
  },
});

export interface AlertDialogProps extends React.HTMLAttributes<HTMLElement>, Pick<BoxProps, 'element'> {
  children: NonNullable<React.ReactNode>;
  destructive?: boolean;
  heading: string;
  isOpen: boolean;
  onConfirm: () => void;
  onConfirmLabel: string;
  onDismiss: () => void;
  onDismissLabel: string;
}

export const AlertDialog = React.forwardRef<HTMLDivElement, AlertDialogProps>(
  (
    {
      children,
      destructive,
      element = 'ALERT_DIALOG',
      heading,
      isOpen,
      onConfirm,
      onConfirmLabel,
      onDismiss,
      onDismissLabel,
      ...props
    },
    ref
  ) => {
    const transitions = useTransition(isOpen, getAnimationStates());
    const headingID = useUID();
    const bodyID = useUID();

    return (
      <>
        {transitions(
          (styles, item) =>
            item && (
              <ModalDialogOverlay isOpen={isOpen} style={{opacity: styles.opacity}}>
                <Box
                  // @ts-expect-error Render overlay as box for customization
                  as={AlertDialogContent}
                  {...safelySpreadBoxProps(props)}
                  aria-labelledby={headingID}
                  aria-describedby={bodyID}
                  element={element}
                  ref={ref}
                  role="alertdialog"
                  style={styles}
                >
                  <AlertDialogHeader headingID={headingID} element={`${element}_HEADER`}>
                    {heading}
                  </AlertDialogHeader>
                  <AlertDialogBody bodyID={bodyID} element={`${element}_BODY`}>
                    {children}
                  </AlertDialogBody>
                  <AlertDialogFooter
                    destructive={destructive}
                    element={`${element}_FOOTER`}
                    onDismiss={onDismiss}
                    onDismissLabel={onDismissLabel}
                    onConfirm={onConfirm}
                    onConfirmLabel={onConfirmLabel}
                  />
                </Box>
              </ModalDialogOverlay>
            )
        )}
      </>
    );
  }
);

AlertDialog.displayName = 'AlertDialog';

AlertDialog.propTypes = {
  children: PropTypes.node.isRequired,
  destructive: PropTypes.bool,
  element: PropTypes.string,
  heading: PropTypes.string.isRequired,
  isOpen: PropTypes.bool.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onConfirmLabel: PropTypes.string.isRequired,
  onDismiss: PropTypes.func.isRequired,
  onDismissLabel: PropTypes.string.isRequired,
};
