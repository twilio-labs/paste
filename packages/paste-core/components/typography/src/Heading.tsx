import * as React from 'react';
import {Text} from '@twilio-paste/text';

export type asTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'div' | 'label' | 'span';
export type HeadingStyle =
  | 'headingStyle10'
  | 'headingStyle20'
  | 'headingStyle30'
  | 'headingStyle40'
  | 'headingStyle50'
  | 'headingStyle60';

interface Heading {
  as: asTags;
  id?: never;
  className?: never;
  children: NonNullable<React.ReactNode>;
  headingStyle?: HeadingStyle;
}

function getHeadingStyles(headingStyle?: HeadingStyle): {} {
  switch (headingStyle) {
    case 'headingStyle60':
      return {
        marginBottom: 'space70',
        fontSize: 'fontSize90',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight60',
      };
    case 'headingStyle50':
    default:
      return {
        marginBottom: 'space60',
        fontSize: 'fontSize70',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight50',
      };
    case 'headingStyle40':
      return {
        marginBottom: 'space50',
        fontSize: 'fontSize60',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight70',
      };
    case 'headingStyle30':
      return {
        marginBottom: 'space40',
        fontSize: 'fontSize40',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight50',
      };
    case 'headingStyle20':
      return {
        marginBottom: 'space30',
        fontSize: 'fontSize30',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight20',
      };
    case 'headingStyle10':
      return {
        marginBottom: 'space30',
        fontSize: 'fontSize20',
        fontWeight: 'fontWeightSemibold',
        lineHeight: 'lineHeight10',
      };
  }
}

const handlePropValidation = ({as, children}: Heading): void => {
  const hasAs = as;
  if (
    hasAs &&
    !(
      as === 'h1' ||
      as === 'h2' ||
      as === 'h3' ||
      as === 'h4' ||
      as === 'h5' ||
      as === 'h6' ||
      as === 'div' ||
      as === 'label' ||
      as === 'span'
    )
  ) {
    throw new Error(`[Paste: Typography Heading] As must be h1, h2, h3, h4, h5, h6, div, label, or span`);
  }

  if (children == null) {
    throw new Error(`[Paste: Typography Heading] Must have non-null children.`);
  }
};

const Heading: React.FC<Heading> = ({as, headingStyle, children}) => {
  handlePropValidation({as, headingStyle, children});

  return (
    <Text as={as} textColor="colorText" {...getHeadingStyles(headingStyle)}>
      {children}
    </Text>
  );
};

Heading.defaultProps = {
  as: 'h2',
  headingStyle: 'headingStyle50',
};

export {Heading};
