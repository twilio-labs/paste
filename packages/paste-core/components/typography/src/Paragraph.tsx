import * as React from 'react';
import {Text} from '@twilio-paste/text';

interface Paragraph {
  id?: never;
  className?: never;
  children: NonNullable<React.ReactNode>;
}

const handlePropValidation = ({children}: Paragraph): void => {
  if (children == null) {
    throw new Error(`[Paste: Typography Paragraph] Must have non-null children.`);
  }
};

const Paragraph: React.FC<Paragraph> = props => {
  handlePropValidation(props);

  return (
    <Text
      as="p"
      marginBottom="space70"
      fontSize="fontSize20"
      lineHeight="lineHeight20"
      fontWeight="fontWeightNormal"
      textColor="colorText"
    >
      {props.children}
    </Text>
  );
};

Paragraph.displayName = 'Paragraph';
export {Paragraph};
