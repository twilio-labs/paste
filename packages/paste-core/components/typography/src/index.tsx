export {Heading} from '@twilio-paste/heading';
export {Paragraph} from '@twilio-paste/paragraph';
