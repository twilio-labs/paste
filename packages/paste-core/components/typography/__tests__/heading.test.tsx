import * as React from 'react';
import renderer from 'react-test-renderer';
import {Theme} from '@twilio-paste/theme';
import {Heading} from '../src';

describe('Heading', () => {
  it('it should render an H1 with 32px font-size', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider>
          <Heading as="h1" headingStyle="headingStyle60">
            This is an H1
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('it should render an H2 with 24px font-size', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider>
          <Heading as="h2" headingStyle="headingStyle50">
            This is an H2
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('it should render an H3 with 20px font-size', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider>
          <Heading as="h3" headingStyle="headingStyle40">
            This is an H3
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('it should render an H4 with 16px font-size', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider>
          <Heading as="h4" headingStyle="headingStyle30">
            This is an H4
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('it should render an H5 with 14px font-size', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider>
          <Heading as="h5" headingStyle="headingStyle20">
            This is an H5
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('it should render an H6 with 12px font-size', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider>
          <Heading as="h6" headingStyle="headingStyle10">
            This is an H6
          </Heading>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
