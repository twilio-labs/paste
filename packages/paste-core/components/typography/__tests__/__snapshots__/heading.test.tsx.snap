// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading it should render an H1 with 32px font-size 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.15;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  font-size: 2rem;
  color: #282a2b;
  font-weight: 600;
  line-height: 1.5;
  margin-bottom: 1.5rem;
}

<div
  className="emotion-1"
>
  <h1
    className="emotion-0"
    fontSize="fontSize90"
    fontWeight="fontWeightSemibold"
  >
    This is an H1
  </h1>
</div>
`;

exports[`Heading it should render an H2 with 24px font-size 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.15;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  font-size: 1.5rem;
  color: #282a2b;
  font-weight: 600;
  line-height: 1.6;
  margin-bottom: 1.25rem;
}

<div
  className="emotion-1"
>
  <h2
    className="emotion-0"
    fontSize="fontSize70"
    fontWeight="fontWeightSemibold"
  >
    This is an H2
  </h2>
</div>
`;

exports[`Heading it should render an H3 with 20px font-size 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.15;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  font-size: 1.25rem;
  color: #282a2b;
  font-weight: 600;
  line-height: 1.5;
  margin-bottom: 1rem;
}

<div
  className="emotion-1"
>
  <h3
    className="emotion-0"
    fontSize="fontSize60"
    fontWeight="fontWeightSemibold"
  >
    This is an H3
  </h3>
</div>
`;

exports[`Heading it should render an H4 with 16px font-size 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.15;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  font-size: 1rem;
  color: #282a2b;
  font-weight: 600;
  line-height: 1.6;
  margin-bottom: 0.75rem;
}

<div
  className="emotion-1"
>
  <h4
    className="emotion-0"
    fontSize="fontSize40"
    fontWeight="fontWeightSemibold"
  >
    This is an H4
  </h4>
</div>
`;

exports[`Heading it should render an H5 with 14px font-size 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.15;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  font-size: 0.875rem;
  color: #282a2b;
  font-weight: 600;
  line-height: 1.714285714;
  margin-bottom: 0.5rem;
}

<div
  className="emotion-1"
>
  <h5
    className="emotion-0"
    fontSize="fontSize30"
    fontWeight="fontWeightSemibold"
  >
    This is an H5
  </h5>
</div>
`;

exports[`Heading it should render an H6 with 12px font-size 1`] = `
.emotion-1 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.15;
  color: #282a2b;
  font-weight: 400;
}

.emotion-1 *,
.emotion-1 *::after,
.emotion-1 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  margin: 0;
  padding: 0;
  font-size: 0.75rem;
  color: #282a2b;
  font-weight: 600;
  line-height: 1.666666667;
  margin-bottom: 0.5rem;
}

<div
  className="emotion-1"
>
  <h6
    className="emotion-0"
    fontSize="fontSize20"
    fontWeight="fontWeightSemibold"
  >
    This is an H6
  </h6>
</div>
`;
